rust/arrow/src/ffi.rs

```diff
@@ -193,6 +204,8 @@ fn to_datatype(format: &str) -> Result<DataType> {
         "ttm" => DataType::Time32(TimeUnit::Millisecond),
         "ttu" => DataType::Time64(TimeUnit::Microsecond),
         "ttn" => DataType::Time64(TimeUnit::Nanosecond),
+        "+l" => DataType::List(Box::new(Default::default())),
+        "+L" => DataType::LargeList(Box::new(Default::default())),
```

Comment:
I am not sure this is sufficient. For nested types, I think that we must transverse the childs before initializing the parent's data type. As is, the parent will get `List(Null)`, but the child will get whatever data type the child has.

We unfortunately do not even validate that `child dataType = parent datatype`, which is why this trick works. However, we will not be fulfilling the specification and can result in UB.

Reply:
I did overwrite them later, however to make it more clear I now added an `Option<DataType>` to indicate the child type. This is provided when `make_array` is called. Internally this is also called by `ArrowArray::buffer_len`. At that moment the child type is not known, but I think it's not important as the child data type is not needed to determine the buffer lengths.